### PR TITLE
Fix override completion with nint/nuint parameters

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/70782.

Issue was a problem in SymbolKey resolution where we weren't handling nints/nuints properly, leading us to think we hadn't properly resolved the prior symbol for the override int he current compilation.  this, in turn, caused us to bail out.

### CODE REVIEW

File: src/Workspaces/CoreTest/SymbolKeyTests.cs [hunk 1]
```diff
@@ -113,6 +114,28 @@ public class B { };
...
+";
+            var netstandardReferences = await ReferenceAssemblies.NetStandard.NetStandard20.ResolveAsync(LanguageNames.CSharp, cancellationToken: default);
+            var netcoreReferences = await ReferenceAssemblies.Net.Net70.ResolveAsync(LanguageNames.CSharp, cancellationToken: default);
+
+            var compilation1 = GetCompilation(source, LanguageNames.CSharp, references: [.. netstandardReferences]);
+            var compilation2 = GetCompilation(source, LanguageNames.CSharp, references: [.. netcoreReferences]);
+            TestRoundTrip(GetDeclaredSymbols(compilation1), compilation1, useSymbolEquivalence: false);
+            TestRoundTrip(GetDeclaredSymbols(compilation1), compilation2, useSymbolEquivalence: true);
+            TestRoundTrip(GetDeclaredSymbols(compilation2), compilation1, useSymbolEquivalence: true);
+            TestRoundTrip(GetDeclaredSymbols(compilation2), compilation2, useSymbolEquivalence: false);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CoreTest/SymbolKeyTests.cs:136]: across the same compilation, they should be .Equals.  Across different compilations, then SymbolEquivalence should match.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs [hunk 2]
```diff
@@ -322,15 +322,30 @@ private bool HandleNamedTypesWorker(INamedTypeSymbol x, INamedTypeSymbol y, Dict
...
+                // If one is a tuple, both must be tuples.
                 if (x.IsTupleType != y.IsTupleType)
                     return false;
 
+                // If one is nint/nuint, the other must bas as well.
                 if (x.IsNativeIntegerType != y.IsNativeIntegerType)
                     return false;
 
+                // If one is void, the other must be as well
+                if (x.IsSystemVoid() != y.IsSystemVoid())
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs:334]: another issue i noticed.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs [hunk 3]
```diff
@@ -322,15 +322,30 @@ private bool HandleNamedTypesWorker(INamedTypeSymbol x, INamedTypeSymbol y, Dict
             {
                 Debug.Assert(GetTypeKind(x) == GetTypeKind(y));
 
+                // If one is a tuple, both must be tuples.
                 if (x.IsTupleType != y.IsTupleType)
                     return false;
 
+                // If one is nint/nuint, the other must bas as well.
```
Comment (dibarbet) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs:329]:
```suggestion
                // If one is nint/nuint, the other must be as well.
```

**Review by dibarbet — APPROVED** [review]

**Review by genlu — APPROVED** [review]
:shipit:

**Pull request merged**